File: js/creational/prototype.js
class Cookie {
    constructor(flavor) {
        this.flavor = flavor
    }

    describe() {
        console.log(`cookie with ${this.flavor} flavor`)
    }

    clone() {
        return new Cookie(this.flavor)
    }
}

const chocoCookie = new Cookie('chocolate')
chocoCookie.describe()

const chocoCookieClone = chocoCookie.clone()
chocoCookieClone.describe()

chocoCookieClone.flavor = 'vanilla'
chocoCookieClone.describe()

chocoCookie.describe()

File: python/creational/prototype.py
class Cookie:
    def __init__(self, flavor):
        self.flavor = flavor

    def describe(self):
        print(f'cookie with {self.flavor} flavor')

    def clone(self):
        return Cookie(self.flavor)


chocoCookie = Cookie('chocolate')
chocoCookie.describe()

chocoCookieClone = chocoCookie.clone()
chocoCookieClone.describe()

chocoCookieClone.flavor = 'vanilla'
chocoCookieClone.describe()

chocoCookie.describe()
